org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/handler/ZigBeeThingHandler.java

```diff
@@ -174,18 +179,43 @@ private synchronized void doNodeInitialisation() {
 
         logger.debug("{}: Start initialising ZigBee Thing handler", nodeIeeeAddress);
 
+        // Update the general properties
+        // This is required here to allow us to check for a static thing type before we check the dynamic configuration
+        ZigBeeNodePropertyDiscoverer propertyDiscoverer = new ZigBeeNodePropertyDiscoverer();
+        propertyDiscoverer.setProperties(getThing().getProperties());
+        Map<String, String> newProperties = propertyDiscoverer.getProperties(coordinatorHandler, node);
+        updateProperties(newProperties);
+
         // Create the channel factory
         ZigBeeChannelConverterFactory factory = new ZigBeeChannelConverterFactory();
+        List<Channel> nodeChannels;
+
+        // Check the thing-type registry to see if we match this thing with a static definition
+        ZigBeeThingTypeMatcher matcher = new ZigBeeThingTypeMatcher();
```

Comment:
This looks like a pretty bad design, if instances have to be / can be created by anyone. Using services would be much better.

Anyhow, I thought that class was about discovery, why is it used within a thing handler?

Reply:
As above - it doesn't use the thing-type registry any more. This is just a helper class to read the properties file to get the associated thing type UID.

The thing type is set in the thing handler - this is what we discussed a long time back (for ZWave at the time) in order to cope with different discovery issues - I simply used the same system for ZigBee.